docs/tutorials/mastering.md

```diff
@@ -899,199 +871,456 @@ Due to the newgrp command you will need to log out twice as it started a new she
+cat << EOF > main.tf
+module "sdcore-control-plane" {
+  source = "git::https://github.com/canonical/terraform-juju-sdcore-k8s//modules/sdcore-control-plane-k8s"
+
+  model_name   = "control-plane"
+  create_model = false
+
+  amf_config = {
+    external-amf-hostname = "amf.mgmt"
+  }
+  traefik_config = {
+    routing_mode = "subdomain"
+  }
+}
+
+resource "juju_offer" "amf-fiveg-n2" {
+  model            = "control-plane"
+  application_name = module.sdcore-control-plane.amf_app_name
+  endpoint         = module.sdcore-control-plane.fiveg_n2_endpoint
+}
+
+EOF
 ```
 
-Deploy the control plane Juju bundle:
+Initialize Juju Terraform provider:
 
 ```console
-juju deploy sdcore-control-plane-k8s --trust --channel=beta --overlay control-plane-overlay.yaml
+terraform init
 ```
 
-Expose the Software as a Service offer for the AMF.
+Deploy SD-Core Control Plane:
 
 ```console
-juju offer control-plane.amf:fiveg-n2
+terraform apply -auto-approve
 ```
 
-### Checkpoint 3: Does AMF External load balancer service exist ?
+Monitor the status of the deployment:
 
-You should be able to see the AMF External load balancer service in Kubernetes. Log in to the `control-plane` VM and execute the following command:
+```console
+watch -n 1 -c juju status --color --relations
+```
+
+The deployment is ready when all the charms are in the `Active/Idle` state. 
+It is normal for `grafana-agent` to remain in waiting state.
+
+Once the deployment is ready, we will proceed to the configuration part.
+
+Log out of the VM.
+
+Get the IP addresses of the AMF and Traefik LoadBalancer services:
+
+Log in to the `control-plane` VM:
 
 ```console
 multipass shell control-plane
-microk8s.kubectl get services -A | grep LoadBalancer
 ```
 
-This will show output similar to the following, indicating:
+Get LoadBalancer services:
 
-- The AMF is exposed on 10.201.0.52 SCTP port 38412
-- The NMS is exposed on 10.201.0.53 TCP ports 80 and 443
+```console
+microk8s.kubectl get services -A | grep LoadBalancer
+```
 
-These IP addresses came from MetalLB and were configured in the bootstrap step.
+This will show output similar to the following:
 
 ```console
 control-plane    amf-external  LoadBalancer  10.152.183.179  10.201.0.52   38412:30408/SCTP
-control-plane    traefik-k8s   LoadBalancer  10.152.183.28   10.201.0.53   80:32349/TCP,443:31925/TCP
+control-plane    traefik       LoadBalancer  10.152.183.28   10.201.0.53   80:32349/TCP,443:31925/TCP
+```
+
+Note both IPs - in this case `10.201.0.52` for the AMF and `10.201.0.53` for Traefik. 
+We will need them shortly. 
+
+Log out of the VM.
+
+Log into the `juju-controller` VM:
+
+```console
+multipass shell juju-controller
+```
+
+Configure AMF external IP, using the address obtained in the previous step. 
+To do that, edit `amf_config` in the `main.tf` file in the `terraform` directory:
+
+```console
+cd terraform
+```
+
+Updated `amf_config` should look like similar to the below:
+
 ```
+(...)
+module "sdcore-control-plane" {
```

Comment:
```suggestion

```console
(...)
module "sdcore-control-plane" {
```

Reply:
That was intentional. The output is nicer without `console`.